src/mixxxmainwindow.cpp

```diff
@@ -430,6 +430,7 @@ void MixxxMainWindow::initialize() {
     if (CmdlineArgs::Instance().getStartAutoDJ()) {
         qDebug("Enabling Auto DJ from CLI flag.");
         ControlObject::set(ConfigKey("[AutoDJ]", "enabled"), 1.0);
+        ControlObject::set(ConfigKey("[Master],recording"), 1.0);
```

Comment:
I was under the impression you were proposing an independent Record CLI switch.
With this change it would always start recording, which is likely unexpected and undesired for most users.
Ie. it fits your use case but I'm opposed to merging it like this.

Please take a look at how `CmdlineArgs::Instance().getStartAutoDJ()` is implemented and add a separate switch for recording.